/*
 * This file was automatically generated by EvoSuite
 * Sat May 24 19:28:04 GMT 2025
 */

package original;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;
import original.Solution2156;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = false) 
public class Solution2156_ESTest extends Solution2156_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      Solution2156 solution2156_0 = new Solution2156();
      String string0 = solution2156_0.subStrHash("I< #}o-?1YOp$D@OPn", 828, 828, 8, 828);
      assertEquals("Op$D@OPn", string0);
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      Solution2156 solution2156_0 = new Solution2156();
      String string0 = solution2156_0.subStrHash("", 2632, 2632, 0, (-3865));
      assertEquals("", string0);
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      Solution2156 solution2156_0 = new Solution2156();
      // Undeclared exception!
      try { 
        solution2156_0.subStrHash("pehx76CAB", 2590, (-1209), 2265, (-1));
        fail("Expecting exception: StringIndexOutOfBoundsException");
      
      } catch(StringIndexOutOfBoundsException e) {
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      Solution2156 solution2156_0 = new Solution2156();
      // Undeclared exception!
      try { 
        solution2156_0.subStrHash((String) null, 2144802644, 2144802644, 2144802644, 2144802644);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("original.Solution2156", e);
      }
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      Solution2156 solution2156_0 = new Solution2156();
      // Undeclared exception!
      try { 
        solution2156_0.subStrHash("w7u! BU$Zr-0]l", 0, 0, 0, 0);
        fail("Expecting exception: ArithmeticException");
      
      } catch(ArithmeticException e) {
         //
         // / by zero
         //
         verifyException("original.Solution2156", e);
      }
  }

  @Test(timeout = 4000)
  public void test5()  throws Throwable  {
      Solution2156 solution2156_0 = new Solution2156();
      String string0 = solution2156_0.subStrHash("Z*A!}a;2K1kf", 1, 2141, 1, 1);
      assertEquals("a", string0);
  }
}
